app/src/main/java/org/metabrainz/android/presentation/features/search/SearchPagingSource.kt

```diff
@@ -19,18 +19,20 @@ class SearchPagingSource(val mbEntityType: MBEntityType, val query: String) : Pa
         return try {
             val response = service.searchEntity(mbEntityType.entity, query, pageSize, offset)?.string()
             var count = LoadResult.Page.COUNT_UNDEFINED
+            val artistList = JsonParser.parseString(response).asJsonObject.get("artists").asJsonArray
             if (offset == 0) {
                 val responseObject = JsonParser.parseString(response)
                 count = responseObject.asJsonObject.get("count").asInt
             }
             val itemsAfter = if (count == LoadResult.Page.COUNT_UNDEFINED) count
             else (count - offset - pageSize).coerceAtLeast(0)
             // itemsAfter is required to be at least otherwise the current page will be not loaded
-
+            val nextKey = if (artistList.isEmpty)  null
```

Comment:
We need a better solution than this one which has to manually check this through if statements